packages/web3-utils/src/socket_provider.ts

```diff
@@ -333,12 +332,13 @@ export abstract class SocketProvider<
 				return;
 			}
 
-			if (jsonRpc.isBatchResponse(response) || jsonRpc.isResponseWithResult(response)) {
+			if (
+				jsonRpc.isBatchResponse(response) ||
+				jsonRpc.isResponseWithResult(response) ||
+				jsonRpc.isResponseWithError(response)
+			) {
 				this._eventEmitter.emit('message', undefined, response);
 				requestItem.deferredPromise.resolve(response);
-			} else {
-				this._eventEmitter.emit('message', response, undefined);
-				requestItem?.deferredPromise.reject(new ResponseError(response));
```

Comment:
This is up for discussion. I removed this code and the tests from `packages/web3-providers-ws/test/unit/web_socket_provider.test.ts` because I couldn't get an error to actually trigger this code because any error resulting from a failed network request was emitted by [_onError](https://github.com/web3/web3.js/blob/a54d63796e5cde9496af91522737a2418820108f/packages/web3-utils/src/socket_provider.ts#L197) and any RPC response with an error is now being resolved. As the only tests written for this code were for RPC response errors, I don't think this code is needed anymore